tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -4598,7 +4598,7 @@ SpirvEmitter::doExtMatrixElementExpr(const ExtMatrixElementExpr *expr) {
     if (colCount > 1)
       indices.push_back(col);
 
-    if (baseExpr->isGLValue()) {
+    if (!baseInfo->isRValue()) {
```

Comment:
What is difference between `baseExpr->isGLValue()` and `!baseInfo->isRValue()`?
It seems they must be the semantically same?!

Reply:
`baseExpr` is checking the LValue-ness from the AST node, but `baseInfo` is checking the LValue-ness from the SpirvInstruction object. Unfortunately the AST node doesn't always have the right value that we need. Since we are creating a temporary object, it is better to use the SpirvInstruction information.